server/lib/session/migrate-sessions.ts

```diff
@@ -25,6 +25,7 @@ export function migrateSessions() {
         ctx.session.acceptedPrivacyVersion = retrievedUser!.acceptedPrivacyVersion
         ctx.session.acceptedTermsVersion = retrievedUser!.acceptedTermsVersion
         ctx.session.acceptedUsePolicyVersion = retrievedUser!.acceptedUsePolicyVersion
+        ctx.session.locale = retrievedUser!.locale
```

Comment:
This is not the appropriate block to put this in, you've written code that will only add locale info to the session if the session did not have `acceptedPrivacyVersion`

You need to put this in its own if statement (outside of the one you put it in). And then you should actually *test* the code using a session that doesn't have locale info in it yet, instead of just assuming that everything you have written works as you expected

Reply:
Oh yeah, my bad (it was late :d).

Btw, what's the point of the `user` variable in this function? It's initialized to `undefined` and not reassigned anywhere, so I feel like I'm missing something here.

Reply:
The point of `user` is that if we have retrieved the `user` previously in this function (because multiple things needed to be migrated), we don't need to re-run that database query N times.

This function exists to fixup old sessions that did not have all the data on them. Yes you need to have handling for `locale` here.

Reply:
And yes, there is a bug here, `user` should be set to `retrievedUser` in these blocks such that the intention is met, so you should fix that